airflow-core/src/airflow/api_fastapi/core_api/routes/public/extra_links.py

```diff
@@ -87,6 +87,29 @@ def get_extra_links(
     )
     all_extra_links = {link_name: link_url or None for link_name, link_url in sorted(all_extra_link_pairs)}
 
+    from airflow.models.xcom import XComModel
+
+    query = XComModel.get_many(
+        dag_ids=dag_id,
+        run_id=dag_run_id,
+        task_ids=task_id,
+        map_indexes=map_index,
+    ).filter(XComModel.key.like("_link_%"))
```

Comment:
```suggestion
    ).filter(XComModel.key.like(r"\_link\_%"))
```

`_` is a special character in SQL LIKE expressions